plugin/pkg/uniq/uniq.go

```diff
@@ -13,19 +18,23 @@ type item struct {
 }
 
 // New returns a new initialized U.
-func New() U { return U{u: make(map[string]item)} }
+func New() *U { return &U{u: make(map[string]item)} }
 
 // Set sets function f in U under key. If the key already exists
 // it is not overwritten.
-func (u U) Set(key string, f func() error) {
+func (u *U) Set(key string, f func() error) {
+	u.Lock()
+	defer u.Unlock()
 	if _, ok := u.u[key]; ok {
 		return
 	}
 	u.u[key] = item{todo, f}
 }
 
 // SetTodo sets key to 'todo' again.
-func (u U) SetTodo(key string) {
+func (u *U) SetTodo(key string) {
```

Comment:
Why did you change this to  a pointer receiver?

Reply:
Right. a map varlue is a pointer to the head of map. So when copy we copy that pointer. Which make any change in that map works from the original or from the copy.